### PR TITLE
more sign warnings fixes

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

**Comment by ssiloti** [comment]
Why disable netlink on Android? It's rather important that we get network change notifications on Android since obviously they happen rather frequently on that platform.

**Comment by aldenml** [comment]
For what I know, netlink support in android is very limited due to (not only) security concerns. I also see that the headers differ from the linux standard.

Before reverting that change, do you test it in android?

File: src/file.cpp [hunk 1]
```diff
@@ -593,7 +593,7 @@ namespace libtorrent
 	{
 		ec.clear();
 		error_code e;
-		file_status s;
+		file_status s = {};
```
Comment (arvidn) [inline src/file.cpp]: what would you think about adding initializers to the members of this class instead?

**Comment by arvidn** [comment]
In some sense it seems more conservative to leave it in until there's a good reason to remove. And, if it should be removed, I think there should be a comment in ``config.hpp``, where the macro used to be, explaining why it's not enabled for android.

Otherwise I imagine someone will just put it back at some point

**Comment by ssiloti** [comment]
I don't test on Android but I'd rather keep it on unless someone reports that it doesn't work.

File: src/file.cpp (hunk 1 above)
Comment (aldenml) [inline src/file.cpp]: Ok, I will do it

**Comment by aldenml** [comment]
Ok, I will revert the `TORRENT_USE_NETLINK` change.

Thinking about adding something related to android in travis.

File: src/enum_net.cpp [hunk 2]
```diff
@@ -1052,8 +1052,8 @@ namespace libtorrent
 		nl_msg->nlmsg_len = NLMSG_LENGTH(sizeof(rtmsg));
 		nl_msg->nlmsg_type = RTM_GETROUTE;
 		nl_msg->nlmsg_flags = NLM_F_DUMP | NLM_F_REQUEST;
-		nl_msg->nlmsg_seq = seq++;
-		nl_msg->nlmsg_pid = getpid();
+		nl_msg->nlmsg_seq = std::uint32_t(seq++);
```
Comment (arvidn) [inline src/enum_net.cpp]: would it work to make ``seq`` ``uint32_t``?

Comment (aldenml) [inline src/enum_net.cpp]: that would require change in `read_nl_sock`, do you prefer that?

Comment (arvidn) [inline src/enum_net.cpp]: only if it would lower the total amount of casting I suppose

Comment (aldenml) [inline src/enum_net.cpp]: I will remove this cast to speed up the merge

**Comment by arvidn** [comment]
lgtm

**Pull request merged**